cmd/virt-handler/virt-handler.go

```diff
@@ -289,7 +289,7 @@ func (app *virtHandlerApp) Run() {
 	// set log verbosity
 	app.clusterConfig.SetConfigModifiedCallback(app.shouldChangeLogVerbosity)
 
-	migrationProxy := migrationproxy.NewMigrationProxyManager(app.serverTLSConfig, app.clientTLSConfig)
+	migrationProxy := migrationproxy.NewMigrationProxyManager(app.serverTLSConfig, app.clientTLSConfig, app.clusterConfig)
```

Comment:
I wonder if it won't be better just to initialize the proxy with `nil` for the client/server tls config, instead of passing the cluster config into the proxy? Then we won't have to fetch the config and check for every socket.

Reply:
How would I detect when it gets enabled again without a Händler restart?

Reply:
I guess with a callback (SetConfigModifiedCallback) that would do `migrationProxy.UpdateTLSConfig`.
Maybe it's not necessary. Just thought about how to avoid multiple calls to the config.

Reply:
Is there an issue with fetching the value from the config?

Reply:
Not that I'm aware of. My comment was about getting this value from the config several times for every migration. Instead, it could be avoided if we will just store it once (and update it when needed) in the migration proxy manager.

Also, chances are low, but if this value will change during the creation of the proxies, some will be started with a tls and some without.

Reply:
> Not that I'm aware of. My comment was about getting this value from the config several times for every migration. 

Fixed.

> Instead, it could be avoided if we will just store it once (and update it when needed) in the migration proxy manager.

I think callbacks should only be used when not avoidable (e.g. to restart controllers on thread count changes). In general it leads to an increased lock usage or other archtectural complications. The configmap uses very narrow locks and can be queried more than fast enough.